### PR TITLE
Add randomObject util

### LINKED ISSUE #12: Add test utility for generating a random object
There have been a few cases where we've needed to generate an object with random keys & values. Usually this involves creating an empty object, looping a certain number of times, and indexing into the object by a random string to assign some value.

An example implementation might look something like this - 

```TS
function randomObject(keyCount?: number) {
    const randomObject = {};
    if (keyCount == null || keyCount <= 0) {
        keyCount = this.randomNumber(1, 10);
    }

    while (Object.keys(randomObject).length < keyCount) {
        const key = faker.random.uuid();
        if (randomObject[key] != null) {
            continue;
        }
        randomObject[key] = this.randomNumber();
    }

    return randomObject;
}
```

### PR DESCRIPTION
**Pull request by vansosnin**
- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

Fix #12

### CODE REVIEW

File: src/utilities/test-utils.test.ts [hunk 1]
```diff
@@ -107,4 +107,29 @@ describe("TestUtils", () => {
     });
 
     // #endregion randomWord
+
+    // -----------------------------------------------------------------------------------------
+    // #region randomObject
```
Comment (brandongregoryscott) [inline src/utilities/test-utils.test.ts:112]: Nitpick - would you mind alphabetizing this region inside this file?

File: src/utilities/test-utils.ts [hunk 2]
```diff
@@ -35,6 +35,22 @@ const _randomValue = <TValue = any>(obj: any): TValue => obj[_randomKey(obj)];
  */
 const _randomWord = (): string => faker.random.word().split(" ")[0];
 
+/**
+ * Generates random object
+ */
+const _randomObject = (keyCount?: number) => {
```
Comment (brandongregoryscott) [inline src/utilities/test-utils.ts:41]: Nitpick - would you mind alphabetizing this function inside this file?

File: src/utilities/test-utils.ts [hunk 3]
```diff
@@ -46,6 +62,7 @@ export const TestUtils = {
     randomKey: _randomKey,
     randomValue: _randomValue,
     randomWord: _randomWord,
+    randomObject: _randomObject,
```
Comment (brandongregoryscott) [inline src/utilities/test-utils.ts:65]: Nitpick - would you mind alphabetizing this property in this object?

**Review by brandongregoryscott — CHANGES_REQUESTED** [review]
Thanks for adding this, @vansosnin! Just left some minor comments for alphabetization (see our general [`CODE-CONVENTIONS.md`](https://github.com/AndcultureCode/AndcultureCode/blob/main/CODE-CONVENTIONS.md) in our home repo for more info)

**Comment by vansosnin** [comment]
Got it, will fix. BTW, can you add those weird hacktoberfest labels? ☺

File: src/utilities/test-utils.test.ts (hunk 1 above)
Comment (vansosnin) [reply src/utilities/test-utils.test.ts]: Done

File: src/utilities/test-utils.ts (hunk 2 above)
Comment (vansosnin) [reply src/utilities/test-utils.ts]: Done

File: src/utilities/test-utils.ts (hunk 3 above)
Comment (vansosnin) [reply src/utilities/test-utils.ts]: Done

**Review by brandongregoryscott — APPROVED** [review]

**Comment by brandongregoryscott** [comment]
Thanks for the contribution, @vansosnin!

**Comment by brandongregoryscott** [comment]
@all-contributors add @vansosnin for code, test

**Pull request merged**